src/main.cpp

```diff
@@ -474,15 +474,14 @@ DEMUX_PACKET* CInputStreamAdaptive::DemuxRead(void)
 
       if (srHaveData)
       {
-        p->dts = static_cast<double>(sr->DTS() + m_session->GetChapterStartTime());
-        p->pts = static_cast<double>(sr->PTS() + m_session->GetChapterStartTime());
+        sr->SetDemuxPacketSideData(p, m_session);
+        p->dts = static_cast<double>(sr->DTS());
+        p->pts = static_cast<double>(sr->PTS());
         p->duration = static_cast<double>(sr->GetDuration());
         p->iStreamId = sr->GetStreamId();
         p->iGroupId = 0;
         p->iSize = iSize;
         std::memcpy(p->pData, pData, iSize);
-
-        sr->SetDemuxPacketSideData(p, m_session);
```

Comment:
there is a reason why you moved setter `SetDemuxPacketSideData` upper, so before assigning data to the pointer?
if no reason please restore setter after assigning pointer properties

Reply:
just before `sr->SetDemuxPacketSideData(p, m_session);`
please add a note something like:
`//! @todo: on Kodi 21, sending of subtitles packet side data is not more needed to be removed`

Reply:
I had moved it when trying a previous strategy (using this method to set the start pts offset) but no longer needed. Will move back and add note